### PR TITLE
Added elastic sink. audit log service

### PR DESCRIPTION
**Pull request by volodymyr-babak**

### CODE REVIEW

File: application/src/main/resources/thingsboard.yml [hunk 1]
```diff
@@ -317,3 +317,16 @@ audit_log:
...
       "rule": "${AUDIT_LOG_MASK_RULE:W}"
       "plugin": "${AUDIT_LOG_MASK_PLUGIN:W}"
+  sink:
+    # Type of external sink. possible options: none, elasticsearch
+    type: "${AUDIT_LOG_SINK_TYPE:none}"
+    # Name of the index where audit logs stored
+    # Index name could contain next placeholders (not mandatory):
+    # @{TENANT} - substituted by tenant ID
+    # @{DATE} - substituted by current date in YYYY.MM.DD format
+    index_pattern: "${AUDIT_LOG_SINK_INDEX_PATTERN:@{TENANT}_AUDIT_LOG_@{DATE}}"
```
Comment (ashvayka) [inline application/src/main/resources/thingsboard.yml]: Please add also "date_format" pattern to the settings.

File: dao/pom.xml [hunk 2]
```diff
@@ -103,7 +103,12 @@
 		<dependency>
 			<groupId>org.springframework</groupId>
 			<artifactId>spring-tx</artifactId>
-		</dependency>       
+		</dependency>
+        <dependency>
```
Comment (ashvayka) [inline dao/pom.xml]: wrong indentation?

File: dao/src/main/java/org/thingsboard/server/dao/audit/AuditLogServiceImpl.java [hunk 3]
```diff
@@ -295,6 +299,10 @@ private AuditLog createAuditLogEntry(TenantId tenantId,
         futures.add(auditLogDao.saveByTenantIdAndEntityId(auditLogEntry));
         futures.add(auditLogDao.saveByTenantIdAndCustomerId(auditLogEntry));
         futures.add(auditLogDao.saveByTenantIdAndUserId(auditLogEntry));
+
+        // TODO: is this correct place to log action into sink?
```
Comment (ashvayka) [inline dao/src/main/java/org/thingsboard/server/dao/audit/AuditLogServiceImpl.java]: I think this is ok for now. Please remove TODO.

File: dao/pom.xml (hunk 2 above)
Comment (volodymyr-babak) [reply dao/pom.xml]: yes

File: application/src/main/resources/thingsboard.yml (hunk 1 above)
Comment (volodymyr-babak) [reply application/src/main/resources/thingsboard.yml]: done

File: dao/src/main/java/org/thingsboard/server/dao/audit/AuditLogServiceImpl.java (hunk 3 above)
Comment (volodymyr-babak) [reply dao/src/main/java/org/thingsboard/server/dao/audit/AuditLogServiceImpl.java]: done

**Pull request merged**